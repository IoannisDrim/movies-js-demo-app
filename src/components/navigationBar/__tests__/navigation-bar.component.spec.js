import NavigationBarComponent from '../navigation-bar.component';

describe('NavigationBarComponent', () => {
  let router = {
    activeState: 'in-theaters',
  };

  it('should render correctly', async () => {
    const navigationBarComponent = new NavigationBarComponent(router);
    document.body.appendChild(navigationBarComponent.render());
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  it('should set active link correctly', () => {
    const navigationBarComponent = new NavigationBarComponent(router);
    document.body.appendChild(navigationBarComponent.render());
    navigationBarComponent.afterRender();
    expect(document.querySelector('.active-header-link').getAttribute('href')).toBe('#in-theaters');
    navigationBarComponent.unmount();
  });

  it('should set active link correctly', () => {
    const navigationBarComponent = new NavigationBarComponent(router);
    document.body.appendChild(navigationBarComponent.render());
    navigationBarComponent.afterRender();
    expect(document.querySelector('.active-header-link').getAttribute('href')).toBe('#in-theaters');
    navigationBarComponent.unmount();
  });

  it('should change active link on url changes', () => {
    const navigationBarComponent = new NavigationBarComponent(router);
    document.body.appendChild(navigationBarComponent.render());
    navigationBarComponent.afterRender();
    navigationBarComponent.router.activeState = 'search-movies';
    window.dispatchEvent(new Event('hashchange'));

    expect(document.querySelector('.active-header-link').getAttribute('href')).toBe(
      '#search-movies',
    );
    navigationBarComponent.unmount();
  });
});
