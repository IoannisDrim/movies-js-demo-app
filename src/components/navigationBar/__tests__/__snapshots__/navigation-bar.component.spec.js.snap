// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBarComponent should render correctly 1`] = `
"<nav id="headerNav" class="navbar">
  <span id="logo" class="navbar__logo">
    MovieRama
    <img src="[object Object]" width="40" alt="movieRamaLogo">
  </span>
  <div class="navbar__menu-list">
    <ul class="navbar__menu-list__items">
      <li><a href="#in-theaters">Now in Theaters</a></li>
      <li><a href="#search-movies">Movies Search</a></li>
    </ul>
  </div>
</nav>
"
`;
